kernel/src/test/java/org/kframework/compile/NumberSentencesTest.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Runtime Verification, Inc. All Rights Reserved.
+package org.kframework.compile;
+
+import static org.junit.Assert.*;
+import static org.kframework.kore.KORE.*;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import org.kframework.attributes.Att;
+import org.kframework.builtin.BooleanUtils;
+import org.kframework.definition.Rule;
+
+public class NumberSentencesTest {
+
+  private static final List<Att> testAtts =
+      Arrays.asList(
+          Att.empty(),
+          Att.empty().add(Att.ANYWHERE()),
+          Att.empty().add(Att.CONCRETE(), "A"),
+          Att.empty().add(Att.CONCRETE(), "B"),
+          Att.empty().add(Att.NON_EXECUTABLE()),
+          Att.empty().add(Att.OWISE()),
+          Att.empty().add(Att.PRIORITY(), "50"),
+          Att.empty().add(Att.PRIORITY(), "100"),
```

Comment:
Something I noticed here. The priority attribute is supposed to be an integer, but it's a string after outer parsing and isn't converted to an integer until `ModuleToKORE` stringifies the definition. So `NumberSentences` should only see a string here.

Reply:
Nice catch, there are similar pain points with any of the attribute values that require parsing/un-parsing - I'd like to fix it with #4003